core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -218,6 +229,21 @@ private IndexMetaData(String index, long version, State state, Settings settings
         this.settings = settings;
         this.mappings = mappings;
         this.customs = customs;
+        // fill missing slots in activeAllocationIds with empty set if needed
+        ImmutableOpenIntMap.Builder<Set<String>> filledActiveAllocationIds = null;
```

Comment:
at that point, numberOfShards has not been computed yet. I would prefer moving all that precomputing code to builder, do you agree?